subiquity/models/locale.py

```diff
@@ -35,6 +36,23 @@ def __init__(self, chroot_prefix):
     def switch_language(self, code):
         self.selected_language = code
 
+    async def gen_localedef(self) -> None:
+        language, charmap = self.selected_language.split(".")
+        cmd = [
+            "localedef",
+            "-f", charmap,
+            "-i", language,
+            "--",
+            f"{language}.{charmap}",
+        ]
+        await arun_command(cmd, check=True)
+
+    async def try_gen_localedef(self) -> None:
+        try:
+            await self.gen_localedef()
+        except subprocess.CalledProcessError as exc:
+            log.warning("Could not generate locale: %r", exc)
```

Comment:
What state does this leave the system?  Sounds like you ran it but it doesn't always work, does it leave the localedef as half-present or something?

Reply:
In my tests, I've only seen the call to localedef fail when running subiquity as non-root (e.g., dry-run).
As far as I know, `localedef` does not alter any locale from the archive except the one specified. That said, the man page says:

```
       One of the following exit values can be returned by localedef:
       0      Command completed successfully.
       1      Warnings or errors occurred, output files were written.
       4      Errors encountered, no output created.
```
and here's what the glib says about exit code 1:
```
     * Some warnings: Write the output file and exit with status 1 to 
     indicate there may be problems using the output file e.g. missing 
     data that makes it difficult to use
```
I feel like the worst thing that can happen would be an incomplete translation but I don't have a read test-case for it.